### PR TITLE
Refactor elements flow and update tests

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Refactor elements flow and update tests

[stage-17]

## Motivation and Context
Update test coverage.

## How Has This Been Tested?
Tested changes locally; unit tests pass.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

**Comment by alex-deaconu** [comment]
@ezequielc please review. Thanks!

File: web/scripts/purchase/services/svc-stripe.js [hunk 1]
```diff
@@ -12,33 +12,23 @@ angular.module('risevision.apps.purchase')
...
       };
       
       this.initializeStripeElements = function (types, options) {
-        var that = this;
-        this.prepareNewElementsGroup();
-
-        return $q.all(types.map(function (type) {
-          return that.createElement(type, options);
-        }));
+        var _stripeClient;
```
Comment (ezequielc) [inline web/scripts/purchase/services/svc-stripe.js:22]: @alex-deaconu I think `_stripeClient` is never used, right? Thanks!

**Review by ezequielc — APPROVED** [review]
@alex-deaconu Looks good 👍 One minor comment. Thanks

File: web/scripts/purchase/services/svc-stripe.js (hunk 1 above)
Comment (alex-deaconu) [reply web/scripts/purchase/services/svc-stripe.js:22]: @ezequielc oops, I left that in there. I will remove. Thanks!

**Pull request merged**